pkg/asset/operators/operators.go

```diff
@@ -0,0 +1,245 @@
+// Package operators deals with creating assets for all operators to be installed for the cluster
+package operators
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/asset/installconfig"
+	"github.com/openshift/installer/pkg/asset/operators/content"
+	"path/filepath"
+	"text/template"
+)
+
+const (
+	keyIndex  = 0
+	certIndex = 1
+)
+
+// operators generates the dependent operator config.yaml files
+type operators struct {
+	assetStock                Stock
+	installConfig             asset.Asset
+	directory                 string
+	rootCA                    asset.Asset
+	etcdCA                    asset.Asset
+	ingressCertKey            asset.Asset
+	kubeCA                    asset.Asset
+	aggregatorCA              asset.Asset
+	serviceServingCA          asset.Asset
+	clusterAPIServerCertKey   asset.Asset
+	etcdClientCertKey         asset.Asset
+	apiServerCertKey          asset.Asset
+	openshiftAPIServerCertKey asset.Asset
+	apiServerProxyCertKey     asset.Asset
+	adminCertKey              asset.Asset
+	kubeletCertKey            asset.Asset
+	tncCertKey                asset.Asset
+	serviceAccountKeyPair     asset.Asset
+	kubeconfig                asset.Asset
+}
+
+var _ asset.Asset = (*operators)(nil)
+
+type genericData map[string]string
+
+// Name returns a human friendly name for the operator
+func (o *operators) Name() string {
+	return "Common Operators"
+}
+
+// Dependencies returns all of the dependencies directly needed by an
+// operators asset.
+func (o *operators) Dependencies() []asset.Asset {
+	return []asset.Asset{
+		o.installConfig,
+		o.assetStock.KubeCoreOperator(),
+		o.assetStock.NetworkOperator(),
+		o.assetStock.Tnco(),
+		o.assetStock.KubeAddonOperator(),
+		o.assetStock.Mao(),
+		o.rootCA,
+		o.etcdCA,
+		o.ingressCertKey,
+		o.kubeCA,
+		o.aggregatorCA,
+		o.serviceServingCA,
+		o.clusterAPIServerCertKey,
+		o.etcdClientCertKey,
+		o.apiServerCertKey,
+		o.openshiftAPIServerCertKey,
+		o.apiServerProxyCertKey,
+		o.adminCertKey,
+		o.kubeletCertKey,
+		o.tncCertKey,
+		o.serviceAccountKeyPair,
+		o.kubeconfig,
+	}
+}
+
+// Generate generates the respective operator config.yml files
+func (o *operators) Generate(dependencies map[asset.Asset]*asset.State) (*asset.State, error) {
+	//cvo := dependencies[o.assetStock.ClusterVersionOperator()].Contents[0]
+	kco := dependencies[o.assetStock.KubeCoreOperator()].Contents[0]
+	no := dependencies[o.assetStock.NetworkOperator()].Contents[0]
+	tnco := dependencies[o.assetStock.Tnco()].Contents[0]
+	//ingress := dependencies[o.assetStock.IngressOperator()].Contents[0]
+	addon := dependencies[o.assetStock.KubeAddonOperator()].Contents[0]
+	mao := dependencies[o.assetStock.Mao()].Contents[0]
+
+	// kco+no+tnco go to kube-system config map
+	kubeSys, err := configMap("kube-system", "cluster-config-v1", genericData{
```

Comment:
https://github.com/openshift/installer/blob/67c75faf001d883fe8b4945d46624afaaaf2792a/installer/pkg/config-generator/generator.go#L168-L173

This config map is incomplete.

Reply:
Fixed.